infoblox/resource_infoblox_ip_allocation.go

```diff
@@ -406,20 +400,28 @@ func resourceAllocationUpdate(d *schema.ResourceData, m interface{}) (err error)
 		return err
 	}
 
+	if d.HasChange("internal_id") {
+		return fmt.Errorf("changing the value of 'internal_id' field is not allowed")
```

Comment:
Can't we used `ForceNew` schema parameter instead?

https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.24.1/helper/schema/schema.go#L111

Reply:
I do not see how this will prevent users from changing the field. ForceNew=true will lead just to resource's destroy-create cycle to be activated, this is not a desired action, most likely. Sometimes this may go even unnoticed among other changes, and at the same time may lead to destruction of the infrastructure's part.